code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -423,19 +423,26 @@ func (fsh *StorageHandler) DownloadFile(
 			"couldn't save latest read marker: %v", err)
 	}
 
-	if len(fileref.EncryptedKey) > 0 {
-		if authToken == nil {
-			return nil, errors.New("auth ticket is required to download encrypted file")
-		}
-		// check if client is authorized to download
-		shareInfo, err := reference.GetShareInfo(
+	var shareInfo *reference.ShareInfo
+	if authToken != nil {
+		shareInfo, err = reference.GetShareInfo(
 			ctx,
 			readMarker.ClientID,
 			authToken.FilePathHash,
 		)
-		if err != nil {
-			return nil, errors.New("error during share info lookup in database" + err.Error())
-		} else if shareInfo == nil || shareInfo.Revoked {
+
+		if err == nil && shareInfo.Revoked {
```

Comment:
app might crash if shareInfo is nil?  It is better to check it first.

Reply:
This is exactly what was done in the previous comment. It' can't be nil if err is nil.